File: app/src/main/java/com/pavelprymak/propodcast/model/db/repo/DbRepo.java
package com.pavelprymak.propodcast.model.db.repo;

import androidx.lifecycle.LiveData;

import com.pavelprymak.propodcast.model.db.FavoriteEpisodeEntity;
import com.pavelprymak.propodcast.model.db.FavoritePodcastEntity;

import java.util.List;

public interface DbRepo {
    //Podcasts
    LiveData<List<FavoritePodcastEntity>> getFavoritePodcasts();

    LiveData<FavoritePodcastEntity> getFavoritePodcastById(String id);

    void insertPodcast(FavoritePodcastEntity podcastEntity);

    void updatePodcast(FavoritePodcastEntity podcastEntity);

    void deletePodcastById(String podcastId);

    LiveData<Integer> getPodcastsCount();


    //Episodes
    LiveData<List<FavoriteEpisodeEntity>> getFavoriteEpisodes();

    LiveData<FavoriteEpisodeEntity> getFavoriteEpisodeById(String id);

    void insertEpisode(FavoriteEpisodeEntity episodeEntity);

    void updateEpisode(FavoriteEpisodeEntity episodeEntity);

    void deleteEpisodeById(String episodeId);

    LiveData<Integer> getEpisodesCount();

}

File: app/src/main/java/com/pavelprymak/propodcast/model/db/repo/DbRepo.kt
package com.pavelprymak.propodcast.model.db.repo

import androidx.lifecycle.LiveData

import com.pavelprymak.propodcast.model.db.FavoriteEpisodeEntity
import com.pavelprymak.propodcast.model.db.FavoritePodcastEntity

interface DbRepo {
    //Podcasts
    fun getFavoritePodcasts(): LiveData<List<FavoritePodcastEntity>>

    fun getPodcastsCount(): LiveData<Int>


    //Episodes
    fun getFavoriteEpisodes(): LiveData<List<FavoriteEpisodeEntity>>

    fun getEpisodesCount(): LiveData<Int>

    fun getFavoritePodcastById(id: String): LiveData<FavoritePodcastEntity>

    fun insertPodcast(podcastEntity: FavoritePodcastEntity)

    fun updatePodcast(podcastEntity: FavoritePodcastEntity)

    fun deletePodcastById(podcastId: String)

    fun getFavoriteEpisodeById(id: String): LiveData<FavoriteEpisodeEntity>

    fun insertEpisode(episodeEntity: FavoriteEpisodeEntity)

    fun updateEpisode(episodeEntity: FavoriteEpisodeEntity)

    fun deleteEpisodeById(episodeId: String)

}
